src/backend/utils/adt/jsonb_util.c

```diff
@@ -585,7 +585,7 @@ pushJsonbValueScalar(JsonbParseState **pstate, JsonbIteratorToken seq,
 			appendKey(*pstate, scalarVal);
 			break;
 		case WJB_VALUE:
-			Assert(IsAJsonbScalar(scalarVal));
+			Assert(scalarVal && IsAJsonbScalar(scalarVal));
```

Comment:
on release build scalarVal is dereferenced in function appendValue

Reply:
Yes, it's better to move the assert there. But what do you think about leaving this as is? IsAJsonbScalar also dereferences scalarVal